### PR TITLE
feat(edge_migration): Add api to allow users to migrate to edge without getting in touch with support

### LINKED ISSUE #1215: Implement self-service Edge API Migrator
- [x] Add button in project settings to migrate from Core to Edge if the project is currently set to Core - should check migration status and report on "Not started", "In Progress", "Complete". We also want to be able to control whether this shows behind a flag, as we want to hide it for some folk with a lot of Identities. 
- [x] Implement trigger API with auth - only org admins can press the button
- [x] Implement Read Migration Status endpoint
- [x] Add copy in project settings with more info and links to docs
- [x] Update docs to show self service button

### PR DESCRIPTION
**Pull request by gagantrivedi**
- Adds an API to send migration event
-  Always generate a migration event (remove sync migration)
-  Add migration status field

### CODE REVIEW

File: api/projects/views.py [hunk 1]
```diff
@@ -103,6 +106,20 @@ def user_permissions(self, request: Request, pk: int = None):
         serializer = UserObjectPermissionsSerializer(instance=permission_data)
         return Response(serializer.data)
 
+    @action(detail=True, methods=["POST"], url_path="migrate-to-edge")
```
Comment (matthewelwell) [inline api/projects/views.py]: Can we add a swagger auto schema decorator here? I guess with the `EmptySerializer` class?

File: api/projects/views.py [hunk 2]
```diff
@@ -103,6 +106,20 @@ def user_permissions(self, request: Request, pk: int = None):
...
+            raise DynamoNotEnabledError()
+
+        project = self.get_object()
+        identity_migrator = IdentityMigrator(project.id)
+
+        if not identity_migrator.can_migrate:
+            raise ProjectMigrationError()
+
+        identity_migrator.start_migration()
+        return Response(status=status.HTTP_202_ACCEPTED)
```
Comment (matthewelwell) [inline api/projects/views.py:130]: Nice use of 202 👍

File: api/environments/dynamodb/migrator.py [hunk 3]
```diff
@@ -29,11 +30,13 @@ def can_migrate(self) -> bool:
         migration_status = self.migration_status
         return migration_status == ProjectIdentityMigrationStatus.MIGRATION_NOT_STARTED
 
+    def start_migration(self):
```
Comment (matthewelwell) [inline api/environments/dynamodb/migrator.py:33]: I guess there's a chance now that we'll end up with some locked projects that failed migration but can't be migrated again, right? That's probably ok but might be worth adding a comment?

**Review by matthewelwell — COMMENTED** [review]
Should we add a constant for the maximum number of identities we allow for a project to migrate themselves? Otherwise we might be opening ourselves up to a heavy load on the database at inopportune times?

File: api/environments/dynamodb/migrator.py (hunk 3 above)
Comment (gagantrivedi) [reply api/environments/dynamodb/migrator.py:33]: Yep, good shout; I have added the comment.

File: api/projects/serializers.py [hunk 4]
```diff
@@ -25,18 +29,20 @@ class Meta:
...
-    def get_use_edge_identities(self, obj: Project) -> bool:
-        if settings.PROJECT_METADATA_TABLE_NAME_DYNAMO:
-            identity_migrator = IdentityMigrator(obj.id)
-            return identity_migrator.is_migration_done or (
-                settings.EDGE_RELEASE_DATETIME
-                and obj.created_date >= settings.EDGE_RELEASE_DATETIME
-            )
+    def get_migration_status(self, obj: Project) -> str:
+        if not settings.PROJECT_METADATA_TABLE_NAME_DYNAMO:
+            return ProjectIdentityMigrationStatus.MIGRATION_NOT_STARTED.value
```
Comment (matthewelwell) [inline api/projects/serializers.py]: Should this be `None` or perhaps we add another value to the enum? `NOT_APPLICABLE` or something?

File: api/projects/serializers.py [hunk 5]
```diff
@@ -25,18 +29,20 @@ class Meta:
...
+        if (
+            settings.EDGE_RELEASE_DATETIME
```
Comment (matthewelwell) [inline api/projects/serializers.py]: We should add this as a property on the Project model if it doesn't already exist?

File: api/projects/serializers.py [hunk 6]
```diff
@@ -25,18 +29,20 @@ class Meta:
             "organisation",
             "hide_disabled_flags",
             "enable_dynamo_db",
-            "use_edge_identities",
```
Comment (matthewelwell) [inline api/projects/serializers.py:28]: Hmm... this is going to be a breaking change on the frontend. Can we leave both fields in for now?

File: api/projects/views.py [hunk 7]
```diff
@@ -106,7 +107,12 @@ def user_permissions(self, request: Request, pk: int = None):
         serializer = UserObjectPermissionsSerializer(instance=permission_data)
         return Response(serializer.data)
 
-    @action(detail=True, methods=["POST"], url_path="migrate-to-edge")
+    @swagger_auto_schema(responses={202: EmptySerializer()}, request_body=no_body)
```
Comment (matthewelwell) [inline api/projects/views.py]: Oh nice, didn't know about the `no_body` util, that's cool! I wonder if there's something similar for the response and we can get rid of the EmptySerializer class entirely?

File: api/projects/views.py [hunk 8]
```diff
@@ -80,7 +81,7 @@ def environments(self, request, pk):
         environments = project.environments.all()
         return Response(EnvironmentSerializerLight(environments, many=True).data)
 
-    @swagger_auto_schema(responses={200: PermissionModelSerializer})
+    @swagger_auto_schema(responses={200: PermissionModelSerializer}, request_body=None)
```
Comment (matthewelwell) [inline api/projects/views.py]: Should we use the `no_body` util here too?

File: api/sales_dashboard/views.py [hunk 9]
```diff
@@ -34,8 +32,6 @@
 
 OBJECTS_PER_PAGE = 50
 
-MAX_MIGRATABLE_IDENTITIES_SYNC = 1000
```
Comment (matthewelwell) [inline api/sales_dashboard/views.py:37]: I still think we should have an upper limit in the code. For self serve migrations at least. @dabeeeenster I know you said we could have a segment but I think we should 'belt and braces' it tbh.

**Review by matthewelwell — APPROVED** [review]
Approved with one comment surrounding a hardcoded upper limit on number of identities.

**Pull request merged**